x-pack/plugins/console_extensions/server/routes/api/text_object/update.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { RequestHandler } from 'kibana/server';
+import { APP } from '../../../../common/constants';
+import { TEXT_OBJECT } from './constants';
+import { RouteDependencies, HandlerDependencies } from './types';
+import { withCurrentUsername } from './with_current_username';
+import { TextObjectSchemaWithId, textObjectSchemaWithId } from './text_object';
+
+export const createHandler = ({ getInternalSavedObjectsClient }: HandlerDependencies) => (
+  username: string
+): RequestHandler<unknown, unknown, TextObjectSchemaWithId> => async (ctx, request, response) => {
+  const client = getInternalSavedObjectsClient();
+  const { id, ...rest } = request.body;
+  try {
+    await client.update(TEXT_OBJECT.type, id, {
```

Comment:
If someone can guess the `id` of another user's textObject, they can overwrite it.

Reply:
Good point! The update operation needs to be scoped to a user id too. It looks like we need to do a read operation (id + username) first, then update.